test/e2e/network/loadbalancer.go

```diff
@@ -1236,6 +1244,143 @@ var _ = common.SIGDescribe("LoadBalancers", func() {
+
+		ginkgo.By(fmt.Sprintf("Creating DaemonSet %q", name))
+		ds, err = cs.AppsV1().DaemonSets(ns).Create(context.TODO(), ds, metav1.CreateOptions{})
+		framework.ExpectNoError(err)
+
+		ginkgo.By("Checking that daemon pods launch on every schedulable node of the cluster")
+		err = wait.PollImmediate(dsRetryPeriod, dsRetryTimeout, e2edaemonset.CheckDaemonPodOnNodes(f, ds, nodeNames))
+		framework.ExpectNoError(err, "error waiting for daemon pods to start")
+		err = e2edaemonset.CheckDaemonStatus(f, name)
+		framework.ExpectNoError(err)
+
+		ginkgo.By(fmt.Sprintf("Creating a service %s with type=LoadBalancer in namespace %s", name, ns))
+		jig := e2eservice.NewTestJig(cs, ns, name)
+		jig.Labels = labels
+		service, err := jig.CreateLoadBalancerService(e2eservice.GetServiceLoadBalancerCreationTimeout(cs), func(svc *v1.Service) {})
+		framework.ExpectNoError(err)
+
+		lbNameOrAddress := e2eservice.GetIngressPoint(&service.Status.LoadBalancer.Ingress[0])
+		svcPort := int(service.Spec.Ports[0].Port)
+
+		ginkgo.By("Hitting the DaemonSet's pods through the service's load balancer")
+		timeout := e2eservice.LoadBalancerLagTimeoutDefault
+		if framework.ProviderIs("aws") {
+			timeout = e2eservice.LoadBalancerLagTimeoutAWS
+		}
+		e2eservice.TestReachableHTTP(lbNameOrAddress, svcPort, timeout)
+
+		ginkgo.By("Starting a goroutine to continuously hit the DaemonSet's pods through the service's load balancer")
+		var totalRequests uint64 = 0
+		var networkErrors uint64 = 0
+		var httpErrors uint64 = 0
+		done := make(chan struct{})
+		defer close(done)
+		go func() {
+			defer ginkgo.GinkgoRecover()
+
+			wait.Until(func() {
+				atomic.AddUint64(&totalRequests, 1)
+				client := &http.Client{
+					Transport: utilnet.SetTransportDefaults(&http.Transport{
+						DisableKeepAlives: true,
+					}),
+					Timeout: 5 * time.Second,
+				}
+				ipPort := net.JoinHostPort(lbNameOrAddress, strconv.Itoa(svcPort))
+				msg := "hello"
+				url := fmt.Sprintf("http://%s/echo?msg=%s", ipPort, msg)
+				resp, err := client.Get(url)
+				if err != nil {
+					framework.Logf("Got error testing for reachability of %s: %v", url, err)
+					atomic.AddUint64(&networkErrors, 1)
+					return
+				}
+				defer resp.Body.Close()
+				if resp.StatusCode != http.StatusOK {
+					framework.Logf("Got bad status code: %d", resp.StatusCode)
+					atomic.AddUint64(&httpErrors, 1)
+					return
+				}
+				body, err := io.ReadAll(resp.Body)
+				if err != nil {
+					framework.Logf("Got error reading HTTP body: %v", err)
+					atomic.AddUint64(&httpErrors, 1)
+					return
+				}
+				if string(body) != msg {
+					framework.Logf("The response body does not contain expected string %s", string(body))
+					atomic.AddUint64(&httpErrors, 1)
+					return
+				}
+			}, time.Duration(0), done)
+		}()
+
+		ginkgo.By("Triggering DaemonSet rolling update several times")
+		var minSuccessRate float64 = 0.95
```

Comment:
Can we add a comment here for why it's 95% for now and a TODO to increase this over time?

Reply:
(based on the current test results, I'm inclined to make this higher like 97% or 98% but will defer to you

Reply:
I suggest let it soak for a bit with 95%, I get burned previously with a similar test from starting  with 99% ...

+100 to a TODO and comment explaining all of this, I was thinking something along these lines "we start with a low but reasonable threshold to analyze the results, but the goal is to achieve 99% , we should do incremental steps toward that goall"

Reply:
I mean, we hit http golang client issues, ephemeral port exhaustion , cloud nat gateway port exhaustion, ... you name it ... if we let it soak for a month we can discard quickly these external factors

Reply:
Updated the PR.

I kept the 95% threshold for now, and added the comment with a `TODO` mention.